### DISCUSSION TITLE
zustand error using Handle tag

### DISCUSSION THREAD

**Discussion by hyosang-park** [xyflow/xyflow#4690] [Q&A] [answered]
Hello, I've been testing some codes in ReactFlow.
Browser shows
[React Flow]: Seems like you have not used zustand provider as an ancestor.
in test code below.

https://codesandbox.io/p/sandbox/f9qz2g

But the error message disappears if I remove Handle tag.

![image](https://github.com/user-attachments/assets/445e3efe-1494-4140-9708-c4d85b5a8557)

Please somebody help me how to use Handle tag without zustand error.

**Comment by bcakmakoglu** [comment]
Your sandbox isn't public.

<img width="653" alt="Screenshot 2024-09-26 at 08 33 57" src="https://github.com/user-attachments/assets/7b0e6eca-2b17-4a6d-b268-56e8bdbf31c2">

**Reply by hyosang-park** [reply]
I didn't know that. I'm sorry. I wonder if the link below works fine.

https://codesandbox.io/p/sandbox/f9qz2g?file=%2Fsrc%2FApp.js

**Comment by hyosang-park** [comment]
```
import React from "react";
import { Handle, Position } from "reactflow";
import ReactFlow, { Background, ReactFlowProvider } from "react-flow-renderer";

const CustomNode = ({ data }) => {
  return (
    <>
      <div style={{ padding: "10px 20px" }}>{data.label}</div>
      <Handle type="target" position={Position.Left} />
    </>
  );
};
const nodeTypes = {
  custom: CustomNode,
};

const FlowChart = () => {
  const initialNodes = [
    {
      id: "1",
      type: "custom",
      data: { label: "Custom Node", name: "Node 1" },
      position: { x: 250, y: 5 },
    },
  ];
  const initialEdges = [];

  return (
    <div>
      <ReactFlow
        nodes={initialNodes}
        edges={initialEdges}
        nodeTypes={nodeTypes}
      >
        <Background color="#aaa" gap={16} />
      </ReactFlow>
    </div>
  );
};

function App(props) {
  return (
    <>
      <ReactFlowProvider>
        <FlowChart />
      </ReactFlowProvider>
    </>
  );
}

export default App;

```

**Reply by bcakmakoglu** [reply, marked as answer]
You're importing from two different packages, causing the issue.

Remove your imports from the rather old package `react-flow-renderer` and replace them with imports from `reactflow`.

**Reply by bcakmakoglu** [reply]
fyi you're also missing styles import `import "reactflow/dist/style.css";`

You can find the v11 docs (for the `reactflow` pkg) here: https://v11.reactflow.dev/

**Reply by hyosang-park** [reply]
Thank you for answering! I owe my escape from the error to you. :)
